parser/src/parser/tests/random_values.rs

```diff
@@ -46,18 +46,15 @@ fn random_values_custom_name() {
         14,
         vec![RandBinding::new(Identifier("$alphas".to_string()), 14)],
     ))]);
-    build_parse_test!(source).expect_ast(expected);
+    ParseTest::new().expect_ast(source, expected);
 }
 
 #[test]
 fn random_values_empty_list_error() {
     let source = "
     random_values:
         rand: []";
-    let error = Error::ParseError(ParseError::InvalidRandomValues(
-        "Random Values section cannot be empty".to_string(),
-    ));
-    build_parse_test!(source).expect_error(error)
+    ParseTest::new().expect_diagnostic(source, "random values cannot be empty");
```

Comment:
nit: To be clearer that we're talking about the section, should this be "random_values cannot be empty" instead? (This would be consistent with what you did below in the "only one declaration may appear in random_values" error). Also, in the case of the trace columns it does actually say "trace_columns section cannot be empty", so maybe we should just follow the same thing here

Reply:
I think the reason I did it this way was because there is the `random_values` section, and then a declaration within that section that defines the random values - it is the latter which the error is referring to, so I wanted to avoid a confusing error message (because in this scenario, the random_values section isn't empty, the declaration inside it is). The actual emitted diagnostic would make it clear what we're referring to, but the message would appear to conflict with the source span shown.

Reply:
ah got it, thanks